File: Problems 3400-3499/3403. Find the Lexicographically Largest String From the Box I/solution.cpp
class Solution {
    public:
        string answerString(string word, int numFriends) {
            if (numFriends == 1) return word;
            int n = word.length();
            int m = n - numFriends + 1;
            char maxc = *max_element(word.begin(), word.end());
            string res;
            for (int i = 0; i < n; i++) {
                if (word[i] != maxc) continue;
                string cand = word.substr(i, m);
                if (cand > res) {
                    res = cand;
                }
            }
            return res;
        }
    };

File: Problems 3400-3499/3403. Find the Lexicographically Largest String From the Box I/solution.py
class Solution:
    def answerString(self, word: str, numFriends: int) -> str:
        # Find lexicographically largest substring where
        # len of substring is no larger than len(word) - numFriends + 1
        # Longer substring will result in larger lexicograph if same
        # prefix
        n = len(word)
        m = n - numFriends + 1
        if numFriends == 1:
            return word
        
        return max(word[i:i+m] for i in range(n))